index.js

```diff
@@ -171,9 +167,9 @@ module.exports.uninstall = function (target, dest, opts) {
             // set the directory where npm uninstall will be run
             opts.cwd = dest;
 
-            // if user added --save flag, pass it to npm uninstall command
+            // if user added --save flag, pass --save-dev flag to npm uninstall command
             if (opts.save) {
-                fetchArgs.push('--save');
+                fetchArgs.push('--save-dev');
```

Comment:
I think we shouldn't change this at all. `npm uninstall --save foo` will uninstall `foo` and remove it from either `dependencies` or `devDependencies`, wherever it was included.

Reply:
Will it?

As per npm CLI documentation, it says:

> npm uninstall takes 3 exclusive, optional flags which save or update the package version in your main package.json:
> 
> -S, --save: Package will be removed from your dependencies.
> 
> -D, --save-dev: Package will be removed from your devDependencies.
> 
> -O, --save-optional: Package will be removed from your optionalDependencies.
> 
> --no-save: Package will not be removed from your package.json file.

Anyway, since we expect Cordova packages to be installed only as a devDependencies, I don't think there should be a problem.

Reply:
I just ran a quick test myself and it does what @raphinesse says.

```
npm init
npm install -D cordova-common
npm uninstall --save cordova-common
```

The example above installs `cordova-common` as a `devDependencies`. `uninstall --save` removed `cordova-common` from `devDependencies` as well.

For a different test case:

```
npm init
npm install cordova-common
npm uninstall --save-dev cordova-common
```

The example above installs `cordova-common` not as a `devDependencies`. `uninstall --save-dev` removed `cordova-common` from `dependencies` as well even though the `--save-dev` flag was passed.

Basically, it seems the flag does make a difference.

Reply:
I have never specified the kind of dependency when uninstalling packages. So yah, we could remove the flag from the uninstall command altogether

Reply:
- Removing the flag might be unsafe as users could have specified 'no-save' in their `.npmrc`
- When I first read the code I found `uninstall --save-dev` to be irritating. I was unsure of it's behavior
- Then again, according to the docs cited by @GedasGa, we definitely _should_ have `--save-dev` and not `--save` here. Basically npm _could_ change the behavior observed by us in a patch release. I don't think they will, but still...
- But what we _actually want_ is the current behavior of `npm`: uninstalling and removing the package from `package.json` regardless of the dependency category. That way, old dependencies that might still be production ones are correctly removed too

So in conclusion, what does that mean? :sweat_smile:

Reply:
I think the way you have it now is correct, @GedasGa :+1: